libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1504,21 +1541,26 @@ void PbftManager::pushSyncedPbftBlocksIntoChain_() {
   if (auto net = network_.lock()) {
     auto round = getPbftRound();
     while (syncBlockQueueSize() > 0) {
-      auto sync_block_opt = processSyncBlock();
-      if (!sync_block_opt) continue;
-      auto &sync_block = *sync_block_opt;
-      auto pbft_block_hash = sync_block.pbft_blk->getBlockHash();
+      auto sync_block = processSyncBlock();
+      if (!sync_block.has_value()) {
+        continue;
+      }
+
+      // Pbft block from sync_block was already processed and saved in db
+      if (pbftSyncBlockProcessed(sync_block.value())) {
+        continue;
+      }
```

Comment:
Same thing here. In processSyncBlock(), will check if have added the block in PBFT chain
Don't need add pbftSyncBlockProcessed function